lambda-functions/database-factory-horizontal-scaling/main.go

```diff
@@ -0,0 +1,331 @@
+		if err != nil {
+			log.WithError(err).Error("Failed to send Mattermost error notification")
+		}
+		return
+	}
+	_, err = checkDBClustersScaling(vpcList)
+	if err != nil {
+		log.WithError(err).Error("Unable to check DB cluster scaling")
+		err = sendMattermostErrorNotification(err, "The Database Factory horizontal scaling Lambda failed.")
+		if err != nil {
+			log.WithError(err).Error("Failed to send Mattermost error notification")
+		}
+		return
+	}
+}
+
+func checkEnvVariables() error {
+	if os.Getenv("RDSMultitenantDBClusterNamePrefix") == "" {
+		return errors.Errorf("Environment variable RDSMultitenantDBClusterNamePrefix was not set")
+	}
+	if os.Getenv("RDSMultitenantDBClusterTagPurpose") == "" {
+		return errors.Errorf("Environment variable RDSMultitenantDBClusterTagPurpose was not set")
+	}
+	if os.Getenv("RDSMultitenantDBClusterTagDatabaseType") == "" {
+		return errors.Errorf("Environment variable RDSMultitenantDBClusterTagDatabaseType was not set")
+	}
+	if os.Getenv("CounterLimit") == "" {
+		return errors.Errorf("Environment variable CounterLimit was not set")
+	}
+	if os.Getenv("Environment") == "" {
+		return errors.Errorf("Environment variable Environment was not set")
+	}
+	if os.Getenv("DBInstanceType") == "" {
+		return errors.Errorf("Environment variable DBInstanceType was not set")
+	}
+	if os.Getenv("TerraformApply") == "" {
+		return errors.Errorf("Environment variable TerraformApply was not set")
+	}
+	if os.Getenv("BackupRetentionPeriod") == "" {
+		return errors.Errorf("Environment variable BackupRetentionPeriod was not set")
+	}
+	if os.Getenv("DatabaseFactoryEndpoint") == "" {
+		return errors.Errorf("Environment variable DatabaseFactoryEndpoint was not set")
+	}
+	if os.Getenv("MattermostNotificationsHook") == "" {
+		return errors.Errorf("Environment variable MattermostNotificationsHook was not set")
+	}
+	if os.Getenv("MattermostAlertsHook") == "" {
+		return errors.Errorf("Environment variable MattermostAlertsHook was not set")
+	}
+	return nil
+}
+
+func getVPCs() ([]string, error) {
+	var next *string
+	sess, err := session.NewSession(&aws.Config{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get initiate AWS session")
+	}
+
+	client := ec2.New(sess)
+	filter := []*ec2.Filter{
+		{
+			Name:   aws.String("tag:Purpose"),
+			Values: []*string{aws.String("provisioning")},
+		},
+		{
+			Name:   aws.String("tag:Terraform"),
+			Values: []*string{aws.String("true")},
+		},
+		{
+			Name:   aws.String("tag:Available"),
+			Values: []*string{aws.String("false")},
+		},
+	}
+	var vpcList []string
+	for {
+		vpcs, err := client.DescribeVpcs(&ec2.DescribeVpcsInput{Filters: filter, NextToken: next})
+		if err != nil {
+			return nil, errors.Wrapf(err, "Unable to get Provisioning VPCs")
+		}
+
+		for _, vpc := range vpcs.Vpcs {
+			vpcList = append(vpcList, *vpc.VpcId)
+		}
+
+		if vpcs.NextToken == nil || *vpcs.NextToken == "" {
+			break
+		}
+
+		next = vpcs.NextToken
+
+	}
+	return vpcList, nil
```

Comment:
Do we want to log or error if something happens and we end up getting no VPCs? It's possible that the describe call could return an empty list so I could see this silently failing for a while before we noticed.

Reply:
Initially I thought that we will never face this issue because that will mean that we are not using any of the provisioning VPCs. But this should not be the case in any of our accounts and this error can help us catch sth bigger in our system. So yes I think its a good idea to catch it here. Added ;)